### PR TITLE
Replace custom SDF_* macros with GZ_*

### PR DESCRIPTION
**Pull request by scpeters**
# 🎉 New feature

Reduce duplicate macro definitions

## Summary

Replace custom implementation of `SDF_DEPRECATED` and `SDF_SUPPRESS_DEPRECATED_*` macros with the `GZ_*` macro counterparts.

## Test it

Remove the `SDF_SUPPRESS_*` lines in [InterfaceElements.cc](https://github.com/gazebosim/sdformat/blob/scpeters/more_gz_macros/src/InterfaceElements.cc#L30-L149) and compile to confirm that `SDF_DEPRECATED` macro still works.

## Checklist
- [X] Signed all commits for DCO
- [ ] Added tests
- [ ] Added example and/or tutorial
- [ ] Updated documentation (as needed)
- [ ] Updated migration guide (as needed)
- [ ] Consider updating Python bindings (if the library has them)
- [X] `codecheck` passed (See [contributing](https://ignitionrobotics.org/docs/all/contributing#contributing-code))
- [X] All tests passed (See [test coverage](https://ignitionrobotics.org/docs/all/contributing#test-coverage))
- [ ] While waiting for a review on your PR, please help review [another open pull request](https://github.com/pulls?q=is%3Aopen+is%3Apr+user%3Aignitionrobotics+repo%3Aosrf%2Fsdformat+archived%3Afalse+) to support the maintainers

**Note to maintainers**: Remember to use **Squash-Merge** and edit the commit message to match the pull request summary while retaining `Signed-off-by` messages.

### CODE REVIEW

**Comment by scpeters** [comment]
windows CI doesn't like this

File: include/sdf/InterfaceElements.hh [hunk 1]
```diff
@@ -169,57 +170,57 @@ class SDFORMAT_VISIBLE NestedInclude
   /// not end with a file extension (it will not end with an extension if it
   /// refers to a model package).
   /// \deprecated Use NestedInclude::Uri() instead
-  public: std::string uri SDF_DEPRECATED(12);
+  public: GZ_DEPRECATED(12) std::string uri;
```
Comment (chapulina) [inline include/sdf/InterfaceElements.hh:173]: Should these say 13?

Comment (scpeters) [reply include/sdf/InterfaceElements.hh:173]: oh, these were actually deprecated in fortress; I suppose they can be removed here. Do you mind if I do that in a separate pull request?

Comment (chapulina) [reply include/sdf/InterfaceElements.hh:173]: Oh yeah I didn't pay attention to the diff. SGTM

**Review by chapulina — APPROVED** [review]

**Pull request merged**